### PR TITLE
Linter to avoid csv merge conflicts

### PR DESCRIPTION
**Pull request by PaliC**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #111163

This PR addresses the persistent issue of merge conflicts in the benchmarks/dynamo/ci_expected_accuracy/ directory, specifically those arising from frequently updated CSV files. Based on @malfet  suggestion, the solution implemented adds three spaces between each line in the CSV files. This approach has proven effective in preventing merge conflicts, as evidenced in [D50239634](https://www.internalfb.com/intern/diff/D50239634/). Regardless of these changes the extra new lines should still allow the csvs to be ingested as normal.

If you have access to the diff:
Normally, modifying a line that is later altered in the stack results in a merge conflict during restacking. With this new spacing strategy, lines that are not modified further down the stack will not trigger merge conflicts, achieving our intended outcome.

cc @albanD @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

**Review by malfet — APPROVED** [review]
LGTM, if it passes CI (specifically linter and torchbench tests)

File: tools/linter/adapters/no_merge_conflict_csv_linter.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                original=original,
+                replacement=replacement,
+                description="Run `lintrunner -a` to apply this patch.",
+            )
+        ]
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Format files with ufmt (black + usort).",
```
Comment (kit1980) [inline tools/linter/adapters/no_merge_conflict_csv_linter.py:93]: Is this correct description?

File: tools/linter/adapters/no_merge_conflict_csv_linter.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+        if replacement == original:
+            return []
+
+        return [
+            LintMessage(
+                path=filename,
+                line=None,
+                char=None,
+                code="UFMT",
```
Comment (kit1980) [inline tools/linter/adapters/no_merge_conflict_csv_linter.py:81]: UFMT or a different code?

Comment (huydhn) [reply tools/linter/adapters/no_merge_conflict_csv_linter.py:81]: I guess it should be `MERGE_CONFLICTLESS_CSV`?

File: tools/linter/adapters/no_merge_conflict_csv_linter.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+
+def check_file(filename: str) -> List[LintMessage]:
+
+    with open(filename, "rb") as f:
+        original = f.read().decode("utf-8")
+    replacement = ""
+    with open(filename, 'r') as f:
+        lines = f.readlines()
+        for line in lines:
```
Comment (huydhn) [inline tools/linter/adapters/no_merge_conflict_csv_linter.py:67]: Hmm, is this going to add 3 new lines to the file every time I run lintrunner, so the file would keep growing.  I need to try this out locally :)

Comment (PaliC) [reply tools/linter/adapters/no_merge_conflict_csv_linter.py:67]: It does not do this, at least when I tested this. Strip should return an empty line for the new lines

**Review by huydhn — APPROVED** [review]

**Review by kit1980 — CHANGES_REQUESTED** [review]
Please clean-up unused code.
For example, `format_error_message` and `as_posix` doesn't look like they are used. And some other things.

File: tools/linter/adapters/no_merge_conflict_csv_linter.py (hunk 2 above)
Comment (kit1980) [reply tools/linter/adapters/no_merge_conflict_csv_linter.py:81]: It's still UFMT?

**Pull request closed without merging**